src/ray/core_worker/task_manager.cc

```diff
@@ -181,13 +181,13 @@ void ObjectRefStream::MarkEndOfStream(int64_t item_index,
   if (end_of_stream_index_ != -1) {
     return;
   }
-  // ObjectRefStream should guarantee the max_index_seen_
+  // ObjectRefStream should guarantee the next_index_
   // will always have an object reference to avoid hang.
   // That said, if there was already an index that's bigger than a given
   // end of stream index, we should mark that as the end of stream.
   // It can happen when a task is retried and return less values
   // (e.g., the second retry is failed by an exception or worker failure).
```

Comment:
This section seems lie it also needs to be updated now

Reply:
Hmm yeah good catch. Now if the task returns fewer items on the second try, we will just drop the extra unconsumed items from the first try. @rkooo567 can you confirm that this is OK? Seems we already have a [test](https://github.com/ray-project/ray/blob/c6f373525b8b1285b91dfa7ef3480a8c4e8f233b/python/ray/tests/test_generators.py#L422) for it.